gnomad_qc/v4/subset.py

```diff
@@ -0,0 +1,222 @@
+import argparse
+import logging
+
+import hail as hl
+
+# TODO: include this when we have the sample QC meta HT: from gnomad_qc.v4.resources.meta import meta
+from gnomad_qc.v4.resources.basics import get_gnomad_v4_vds
+
+# TODO: Can remove when we use the full sample QC meta HT
+from gnomad_qc.v4.resources.meta import project_meta
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s: %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+UKB_BATCHES_FOR_INCLUSION = {"100K", "150K", "200K"}
+
+HEADER_DICT = {
+    "format": {
+        "GT": {"Description": "Genotype", "Number": "1", "Type": "String"},
+        "AD": {
+            "Description": "Allelic depths for the ref and alt alleles in the order listed",
+            "Number": "R",
+            "Type": "Integer",
+        },
+        "DP": {
+            "Description": "Approximate read depth",
+            "Number": "1",
+            "Type": "Integer",
+        },
+        "GQ": {
+            "Description": "Phred-scaled confidence that the genotype assignment is correct. Value is the difference between the second lowest PL and the lowest PL (always normalized to 0).",
+            "Number": "1",
+            "Type": "Integer",
+        },
+        "MIN_DP": {
+            "Description": "Minimum DP observed within the GVCF block",
+            "Number": "1",
+            "Type": "Integer",
+        },
+        "PGT": {
+            "Description": "Physical phasing haplotype information, describing how the alternate alleles are phased in relation to one another",
+            "Number": "1",
+            "Type": "String",
+        },
+        "PID": {
+            "Description": "Physical phasing ID information, where each unique ID within a given sample (but not across samples) connects records within a phasing group",
+            "Number": "1",
+            "Type": "String",
+        },
+        "PL": {
+            "Description": "Normalized, phred-scaled likelihoods for genotypes as defined in the VCF specification",
+            "Number": "G",
+            "Type": "Integer",
+        },
+        "SB": {
+            "Description": "Per-sample component statistics which comprise the Fisher's exact test to detect strand bias. Values are: depth of reference allele on forward strand, depth of reference allele on reverse strand, depth of alternate allele on forward strand, depth of alternate allele on reverse strand.",
+            "Number": "4",
+            "Type": "Integer",
+        },
+        "RGQ": {
+            "Description": "Unconditional reference genotype confidence, encoded as a phred quality -10*log10 p(genotype call is wrong)"
+        },
+    },
+}
+
+
+def main(args):
+    hl.init(log="/subset.log", default_reference="GRCh38")
+    header_dict = HEADER_DICT
+    test = args.test
+
+    vds = get_gnomad_v4_vds(n_partitions=args.n_partitions)
```

Comment:
Good call, I forgot that we default to removing samples in the dataset loading call. It makes sense to keep it for most scripts, but yes collaborators will likely want all samples (the previous subsetting script had hard filters as an option but I removed it for the MVP since we don't actually have that info on the full sample set yet and need this ASAP).